src/main/java/com/google/api/generator/gapic/composer/defaultvalue/DefaultValueComposer.java

```diff
@@ -249,8 +254,10 @@ static Expr createResourceHelperValue(
       List<ResourceName> unexaminedResnames = new ArrayList<>(resnames);
       for (ResourceName resname : resnames) {
         unexaminedResnames.remove(resname);
-        if (!resname.isOnlyWildcard()) {
-          return createResourceHelperValue(resname, false, unexaminedResnames, fieldOrMessageName);
+        if (!resname.isOnlyWildcard()
```

Comment:
I see a couple of new logics in this method, and I understand it's in composers so unit testing it might get complicated, can we add some test cases in the protos if it's not too hard?

Reply:
My general feeling about this codebase is that it is beyond comprehension unfortunately and this affects the testing and developing strategy.
This new logic is being tested by the current set of tests already.

Reply:
I understand the sentiment and I totally feel you when I was developing dynamic routing headers, but do we want to make it worse?  e.g. I added this [proto](https://github.com/googleapis/gapic-generator-java/blob/main/src/test/proto/explicit_dynamic_routing_header_testing.proto) to test different combinations for dynamic routing headers, it may still not cover everything but I think it helps.

Reply:
I really believe that those extra tests are not making it better, but making it worse. Basically we add more unmaintainable code. This codebase has 10+ times more code than it should for the task it performs.